src/player/Player.php

```diff
@@ -2135,6 +2138,10 @@ public function respawn() : void{
 		$this->setHealth($this->getMaxHealth());
 
 		foreach($this->attributeMap->getAll() as $attr){
+			if($attr->getId() === Attribute::EXPERIENCE or $attr->getId() === Attribute::EXPERIENCE_LEVEL){ //we have already reset both of those if needed when the player died
+				$this->getNetworkSession()->syncAttributes($this, [$attr]);
```

Comment:
why is this necessary?

Reply:
Because the client on respawn assumes vanilla behaviour and automatically resets experience progress and level to 0. Without this when your respawn it will appear that you have lost your experience but if you leave and join again it appears back.

Reply:
I think markSynchronized(false) should be sufficient.

Reply:
Note to self: this reset might also be causing some other problems (for example setting health to max before resetting attribute defaults might lead to health overflow).